### PR TITLE
[AIRFLOW-6206] Move and rename AWS batch operator [AIP-21]

### PR DESCRIPTION
**Pull request by dazza-codes**
### Jira

- [x] My PR addresses the following [Airflow Jira]
  - https://issues.apache.org/jira/browse/AIRFLOW-6206
  - https://issues.apache.org/jira/browse/AIRFLOW-4733 [AIP-21]

### Description

- [x] basic description

Modules and classes for AWS batch are adapted to https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-21%3A+Changes+in+import+paths

### Tests

- [x] My PR does not change existing test success and it adds deprecation tests

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - existing documentation is patched to use the revised module name

### CODE REVIEW

**Comment by mik-laj** [comment]
Hi. Can you add backward compatibility and move the module to new package `airflow.providers` according to AIP-21? You should also add test cases in https://github.com/apache/airflow/blob/master/tests/test_core_to_contrib.py

**Comment by dazza-codes** [comment]
@mik-laj - With respect, adding those requirements onto this PR seems out of place.  This PR is really only about renaming the module, the additional requirements for AIP-21 seem to belong elsewhere with a larger migration of many contribution modules?  It seems out of place to do that migration just for this one module in this PR.  AFAICT, the use of this module might not be changed because the import patterns might not refer directly to the full package path for this module, but get it via higher level packaging (or wrapping) and that is not changed here in this PR.  If it is a breaking change, then it should be so because the naming pattern should be consistent across all the AWS providers (contrib modules).

Since the batch operator is not already in the https://github.com/apache/airflow/blob/master/tests/test_core_to_contrib.py, it was not found with a git-grep.  It might be an oversight that it's not there already, or an intentional omission.  Is there a clear answer as to why it is not already there?  If it's a simple omission and it's easy to add a two-line mapping, I guess it could be added here in this PR.  FYI, compare the outputs from
```
git grep 'AWSBatchOperator'
git grep 'AWSAthenaOperator'
```
You will see that the `AWSBatchOperator` is not already in the `airflow/providers/amazon/aws/operators/` package and so adding it to that test fails the test.  Based on that, my best guess is that it's an intentional omission from that test.

**Comment by mik-laj** [comment]
This is not just a module change, but your implementation is not backwards compatible, which can cause a problem for users who use this class.
This code will start throwing an exception if your change is accepted.
```
from airflow.contrib.operators.awsbatch_operator import AWSBatchOperator
```
Instead, it should display a deprecation message and work correctly in subsequent versions.
https://github.com/apache/airflow/pull/6455/files
https://github.com/apache/airflow/pull/6454/files

In the file https://github.com/apache/airflow/blob/master/tests/test_core_to_contrib.py we have tests that check whether previous import paths still work after moving the class to a new location. Your service has not been adapted to AIP-21, so it has not been added yet, but if you are moving files then you should move the files to a good place.  The file cannot be moved to the wrong/temporary place, because migration requires the user's intervention and will be very confused if in each version he will have to change the import path.

Can you move also change location of tets?
https://github.com/apache/airflow/blob/master/tests/contrib/operators/test_awsbatch_operator.py

**Comment by darrenleeweber** [comment]
Where are details on AIP-21, I need to RTM on that.

**Comment by dazza-codes** [comment]
Marking this PR as WIP to bring it into line with
- https://github.com/apache/airflow/blob/master/UPDATING.md#migration-of-aws-components
- https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-21%3A+Changes+in+import+paths

At the time of writing this comment, there is ongoing email threads about naming conventions and the location of the new provider packages.  Some comments in those emails are proposing some changes that are inconsistent with the current `master` branch (or at least the UPDATING.md doc), so it's not entirely clear where to go with this PR to align it with AIP-21 exactly.  IMO, this PR stands on it's merits as a stepping stone toward AIP-21, but the additional work to align it could be done here.  I don't understand why it's been overlooked so far in other PRs for AIP-21.

**Comment by dazza-codes** [comment]
The latest commit will be squashed when it's working.  The last piece is to get the core-to-contrib tests all passing, but I'm not understanding the failure for:
```
FAILED tests/test_core_to_contrib.py::TestMovingCoreToContrib::test_is_class_deprecated_7_airflow_providers_amazon_aws_operators_batch_AwsBatchOperator
- AssertionError: DeprecationWarning not triggered
```
The deprecation warning is coded for the module, but the class name is also changing from `AWSBatchOperator` to `AwsBatchOperator` and maybe that it throwing off the parameterized test somehow?  The level of abstraction in the parameterized test is bending my mind a bit too far to immediately grok the problem.

Also, need to update doc links.

File: airflow/contrib/operators/awsbatch_operator.py [hunk 1]
```diff
@@ -17,199 +17,17 @@
...
-
-    def on_kill(self):
-        response = self.client.terminate_job(
-            jobId=self.jobId,
-            reason='Task killed by the user')
-
-        self.log.info(response)
+# pylint: disable=unused-import
+from airflow.providers.amazon.aws.operators.batch import BatchProtocol  # noqa
+from airflow.providers.amazon.aws.operators.batch import AwsBatchOperator as AWSBatchOperator  # noqa
```
Comment (mik-laj) [inline airflow/contrib/operators/awsbatch_operator.py]: If you are changing the class name, you should add a deprecation message in init. Example: https://github.com/apache/airflow/pull/6771/files

Comment (dazza-codes) [reply airflow/contrib/operators/awsbatch_operator.py]: OK thanks, done.

**Comment by dazza-codes** [comment]
Resolved WIP status and review requests to conform with AIP-21 (add the AIP-21 tag to the PR).

**Comment by dazza-codes** [comment]
The CI failed only for one "platform" and it's not related to this PR, i.e. failures were mostly cassandra:
```
SKIPPED [1] tests/sensors/test_sql_sensor.py:72: this is a postgres test
SKIPPED [1] tests/www/test_views.py:1817: flaky when run on mysql
FAILED tests/providers/apache/cassandra/hooks/test_cassandra_hook.py::TestCassandraHook::test_get_conn
FAILED tests/providers/apache/cassandra/hooks/test_cassandra_hook.py::TestCassandraHook::test_get_lb_policy
FAILED tests/providers/apache/cassandra/hooks/test_cassandra_hook.py::TestCassandraHook::test_record_exists_with_keyspace_from_cql
FAILED tests/providers/apache/cassandra/hooks/test_cassandra_hook.py::TestCassandraHook::test_record_exists_with_keyspace_from_session
FAILED tests/providers/apache/cassandra/hooks/test_cassandra_hook.py::TestCassandraHook::test_table_exists_with_keyspace_from_cql
FAILED tests/providers/apache/cassandra/hooks/test_cassandra_hook.py::TestCassandraHook::test_table_exists_with_keyspace_from_session
FAILED tests/providers/apache/cassandra/sensors/test_cassandra_sensor.py::TestCassandraRecordSensor::test_poke
FAILED tests/providers/apache/cassandra/sensors/test_cassandra_sensor.py::TestCassandraTableSensor::test_poke
==== 8 failed, 4037 passed, 115 skipped, 125 warnings in 1190.70s (0:19:50) ====
```
All I can do for that is hit a travis-retry button.  Actually, I don't have travis permissions to retry that particular test plaform for `Tests mysql python 3.7`.

**Comment by dazza-codes** [comment]
I'll try to rebase and resolve conflicts after #6765 was merged (thanks).

**Comment by dazza-codes** [comment]
Rebased and pushed the AIP-21 changes.
- the git diff is not tracking it like a pure file move and might be losing some git history details, so if that's important for git-blame and other tracking, this might need to be split up into several commits for (a) a pure git-mv and (b) subsequent changes (ping me if this needs to happen, otherwise carry on regardless)

@mik-laj - it's ready for another round of review and please note that additional features are going to be stacked on this PR.

**Pull request merged**